### PR TITLE
Add python script to run automated tests

### PR DESCRIPTION
**Pull request by nahiim**

### CODE REVIEW

**Comment by AnastaZIuk** [comment]
**[CI]**: Can one of the admins verify this patch?

**Comment by nahim-yay** [comment]
@Erfan-Ahmadi will review it

File: run_tests.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def run_tests():
+	for i in range(len(indices)):
+		for fn in glob.glob(indices[i] + ".*"):
+		    print("\n", fn)
+		    os.chdir(fn)
+		    os.chdir("bin")
+
+		for fn in glob.glob("*.exe"):
```
Comment (Erfan-Ahmadi) [inline run_tests.py:32]: We don't need every exe in `bin/` we just need the release version and it has the format of `targetname.exe` and NOT `targetname_d.exe` or `targetname_rwdi.exe`
and targetname is `tolower(examplefoldername without the number and the dot at the beginning)` please verify @AnastaZIuk 

Also consider as example failure and log appropriate message when this exe is not found as if the example failed

File: run_tests.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			p = subprocess.Popen(fn, stdout=subprocess.PIPE, shell=False, text=True)
+			try:
+				outs, errs = p.communicate(timeout=10)
+			except subprocess.TimeoutExpired:
```
Comment (Erfan-Ahmadi) [inline run_tests.py:36]: I know this is a temporary solution until we add "CI Mode" to our examples BUT not all of our examples have renderloop until the end of time. there are examples that test something and quit after a short time like `14.ComputeTest`.
That way the example quits but you don't log anything and don't consider it's return code.

Let me rephrase: `the timeout is only there to make sure we don't wait forever for the example to exit, but you must always assume that the example will quit after a short amount of time and consider it's return code`

File: run_tests.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+				p.terminate()
+				p.wait()
+				outs, errs = p.communicate()
+
+				if p.returncode == 0 or p.returncode == 1:
+					return_status = 'SUCCESS'
+				else:
+					return_status = 'FAILURE'
+				print('EXIT STATUS CODE: ', p.returncode, '(', return_status, ')')
```
Comment (Erfan-Ahmadi) [inline run_tests.py:45]: It's best if we first print the result (SUCCESS/FAILIURE) and then the reason (RETURN_CODE, TIMEOUT, ...)
note that now the TIMEOUT is not a reason but it should be in the future when we add CI mode and we expect the example to finish in at most N seconds.

**Comment by nahim-yay** [comment]
-Removed Timeout
-Handle FileNotFoundException
-Log results to html file

File: run_tests.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	for i in range(len(indices)):
+		for fn in glob.glob(indices[i] + ".*"):
+			print("\n", fn)
+			filename = strip_name(fn)
+			print('',filename)
+			os.chdir(fn)
+
+			if "bin" in list(filter(os.path.isdir, os.listdir(os.curdir))):
+				os.chdir("bin")
+				try:
```
[the remark this answers is not in the document]
Comment (devshgraphicsprogramming) [reply run_tests.py:89]: > Succeeded or Failed and what's the error and the you can log and print or write to html just once

actually its best to write the HTML as you go, so if script crashes or something happens you still have a partial result

File: run_tests.py (hunk 1 above)
Comment (AnastaZIuk) [reply run_tests.py]: configuration*, my phone autocorrection is ugly

File: run_tests.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Erfan-Ahmadi) [reply run_tests.py:89]: I'm thinking about the scope of a single example, not do everything at the end. It's basically cleaning code, doesn't change what instructions gets executed much. And it's Python, not like it's gonna crash or anything

[the remark this answers is not in the document]
Comment (devshgraphicsprogramming) [reply run_tests.py:89]: > And it's Python, not like it's gonna crash or anything

an exception is a crash.

Its best to prioritize getting as much useful output from a failed run as possible.

**Comment by nahim-yay** [comment]
Handled Validation Errors and Timeout. Now I'm waiting for review to proceed to next steps which involve CI mode

**Comment by nahim-yay** [comment]
Handled Validation Errors and Timeout. Now I'm waiting for review to proceed to next steps which involve CI mode

File: run_tests.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+import glob, os, subprocess, time
+
+
+# walk into the examples_tests directory
+os.chdir("examples_tests")
+
+# list containing indices of each example to test
+# (I'm supposed to use list comprehension here but for some weird reason 'glob' refuse to work when I use it :\)
+indices = [1,2,3]
```
[the remark this answers is not in the document]
Comment (devshgraphicsprogramming) [reply run_tests.py]: we need to maintain a separate list of what examples to test, not every example folder that exists will be built
